build/builder.go

```diff
@@ -942,6 +945,40 @@ func sortDocuments(todo []*zoekt.Document) {
 	}
 }
 
+// sortDocuments2 sorts zoekt.Document according to their Scores. In general,
+// documents can have a nil score vector if the document to be indexed was
+// introduced after the ranking took place. A nil score vector translates to the
+// lowest possible rank.
+func sortDocuments2(rs []*zoekt.Document) {
+	sort.Slice(rs, func(i, j int) bool {
+		r1 := rs[i].Scores
+		r2 := rs[j].Scores
+
+		if r1 == nil && r2 == nil {
+			return false
+		}
+
+		if r1 == nil {
+			return false
+		}
+
+		if r2 == nil {
+			return true
+		}
+
+		for i := range r1 {
+			if r1[i] < r2[i] {
```

Comment:
you are using the assumping that r1 and r2 have the same length.

Reply:
What an assumption to make!

Reply:
(Sorry, just got a random notification for this comment and felt like saying hi, miss you guys)

Reply:
Hope you are doing well!

Reply:
Yes, I make sure this is the case by checking the invariant in the cilent. We could also change the logic here by ranking shorter vectors higher and allow all kinds lof length. WDYT?

Reply:
👋 @tsenart We miss you too!

Reply:
@tsenart :wave: